File: models/jugador.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Jugador extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      // Un jugador pertenece a un equipo
      Jugador.belongsTo(models.Equipo, { foreignKey: 'equipo_id' });
    }
  }
  Jugador.init({
    nombre: DataTypes.STRING,
    posicion: DataTypes.STRING,
    edad: DataTypes.INTEGER,
    equipo_id: DataTypes.INTEGER
  }, {
    sequelize,
    modelName: 'Jugador',
    tableName: 'jugadors',
    timestamps: true
  });
  return Jugador;
};

File: Flutter/src/nemer_app/lib/models/jugador.dart
class Jugador {
  final int id;
  final String nombre;
  final String posicion;
  final int edad;
  final int equipoId;

  Jugador({
    required this.id,
    required this.nombre,
    required this.posicion,
    required this.edad,
    required this.equipoId,
  });

  factory Jugador.fromJson(Map<String, dynamic> json) {
    return Jugador(
      id: json['id'] ?? 0,
      nombre: json['nombre'] ?? '',
      posicion: json['posicion'] ?? '',
      edad: json['edad'] ?? 0,
      equipoId: json['equipo_id'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "nombre": nombre,
      "posicion": posicion,
      "edad": edad,
      "equipo_id": equipoId,
    };
  }
}
